OSDOCS-2722: Create release note section for Compliance Operator (#37975)
openshift/openshift-docs: security/compliance_operator/compliance-operator-release-notes.adoc

[id="compliance-operator-release-notes-0-1-44"]
== Release Notes for Compliance Operator 0.1.44
Issued: 2021-

The Compliance Operator 0.1.44 is now available with enhancements and bug fixes.

[id="compliance-operator-0-1-44-new-features-and-enhancements"]
=== New features and enhancements
* Add option to make scan scheduling strict/not strict - adds a new ComplianceScan/Suite/ScanSetting option called strictNode scan. This option defaults to true, meaning that the operator will error out of a scan pod can't be scheduled. Switching the option to true makes the scan more permissive and go forward. Useful for clouds with ephemeral nodes.
+
* Result Server: Make nodeSelector and tolerations configurable - exposes the nodeSelector and tolerations attributes through the ScanSettings object. This enables deployers to configure where the Result Server will run, and thus what node will host the Persistent Volume that will contain the raw results. This is needed for cases where the storage driver doesn't allow us to schedule a pod that makes use of a persistent volume on the master nodes.

[id="openshift-compliance-operator-0-1-44-bug-fixes"]
=== Bug fixes

> * Previously, a crash was occurring. In this release, the Compliance Operator now uses uses the latest openscap 1.3.5 to avoid this potential crash.

Reviewer: Total nit, but could we say that previously a crash was occuring in one of the `scanner` containers of the pods performing the scan? Just to make it very clear that it wasn't CO itself that was crashing.

Author: Yes, thanks for the clarification. You might notice some other areas where I took a stab at rewriting, but need some additional info. All comments are welcome.

//OpenShift Compliance Operator Release Notes
[id="compliance-operator-release-notes"]
= Compliance Operator release notes
:context: compliance-operator-release-notes-v0
include::modules/common-attributes.adoc[]

toc::[]

> 

Reviewer 2: Please add an introductory sentence between these headings.

[id="compliance-operator-inclusive-language"]
== Making open source more inclusive

Red Hat is committed to replacing problematic language in our code, documentation, and web properties. We are beginning with these four terms: master, slave, blacklist, and whitelist. Because of the enormity of this endeavor, these changes will be implemented gradually over several upcoming releases. For more details, see link:https://www.redhat.com/en/blog/making-open-source-more-inclusive-eradicating-problematic-language[Red Hat CTO Chris Wright's message].

[id="compliance-operator-release-notes-0-1-44"]
== OpenShift Compliance Operator 0.1.44

The following advisory is available for the OpenShift Compliance Operator 0.1.44:

* link:https://access.redhat.com/errata/RHBA-2021:4530[RHBA-2021:4530 OpenShift Compliance Operator Bug Fix and Enhancement Update]

[id="compliance-operator-0-1-44-new-features-and-enhancements"]
=== New features and enhancements

> * In this release, the `strictNodeScan` option is now added to the `ComplianceScan`, `ComplianceSuite` and `ScanSetting` CRs. This option defaults to "true" which matches the previous behavior where an error occurred if a scan was not able to be scheduled on a node. Setting the option to "false" (meaning not strict, permissive) allows the Compliance Operator to be more lenient about scheduling scans. Environments with ephemeral nodes can set `strictNodeScan` to false which will allow a compliance scan to proceed, even if some of the nodes in the cluster are not available for scheduling.

Reviewer 2: You might want to reconsider the use of parentheses. The ISG suggests that one avoid using them in running text because it means that the text they contain is not very important.

```suggestion
* In this release, the `strictNodeScan` option is now added to the `ComplianceScan`, `ComplianceSuite` and `ScanSetting` CRs. This option defaults to `true`, which matches the previous behavior, where an error occurred if a scan was not able to be scheduled on a node. Setting the option to `false` (meaning not strict, permissive) allows the Compliance Operator to be more lenient about scheduling scans. Environments with ephemeral nodes can set the `strictNodeScan` value to false, which allows a compliance scan to proceed even if some of the nodes in the cluster are not available for scheduling.
```

+ 

> * You can now customize the nodes that are used to schedule the `ResultServer` workload by configuring the `nodeSelector` and `tolerations` attributes of the `ScanSetting` object. These attributes are used to place the `ResultServer` pod (a pod where the raw ARF results are uploaded to, the pod mounts a PV storage volume). Previously, the `nodeSelector` and the `tolerations` defaulted to selecting master nodes and tolerating the node-role.kubernetes.io/master taint. This did not work in environments where master nodes are not permitted to mount PVs. This feature provides another way for you to select nodes and tolerate different taints in those environments.  

Reviewer 2: Double-check the use of "master" here, and please take another look at the text in parentheses. It's confusing and/or a comma splice.

```suggestion
* You can now customize the nodes that are used to schedule the `ResultServer` workload by configuring the `nodeSelector` and `tolerations` attributes of the `ScanSetting` object. These attributes are used to place the `ResultServer` pod (a pod where the raw ARF results are uploaded to, the pod mounts a PV storage volume). Previously, the `nodeSelector` and the `tolerations` parameters defaulted to selecting master nodes and tolerating the `node-role.kubernetes.io/master` taint. This did not work in environments where master nodes are not permitted to mount PVs. This feature provides another way for you to select nodes and tolerate different taints in those environments.  
```

Author: Changes made.

+
* The Compliance Operator can now remediate `KubeletConfig` objects.
+
* A comment containing an error message is now added to help content developers differentiate between objects that do not exist in the cluster versus objects that cannot be fetched.
+

> * Rule objects now contain two new attributes, `checkType` and `description`. These attributes allow you to determine if the rule pertains to a node check or platform check, and also allows you to review what the rule does.

Reviewer 2: ```suggestion
* Rule objects now contain two new attributes, `checkType` and `description`. These attributes allow you to determine if the rule pertains to a node check or platform check and also allow you to review what the rule does.
```

+

> * This enhancement removes the requirement that you have to extend an existing profile in order to create a tailored profile. This means the `extends` field in the `TailoredProfile` CRD is no longer mandatory. You can now select a list of rule objects to create a tailored profile from scratch.  Note: You will need to select whether your profile applies to nodes or the platform by setting the `compliance.openshift.io/product-type:` annotation or by setting the `-node` suffix for the `TailoredProfile` CR.

Reviewer 2: "from scratch" is idiomatic, and, for better internationalization, the ISG says to avoid using idioms.

```suggestion
* This enhancement removes the requirement that you have to extend an existing profile in order to create a tailored profile. This means the `extends` field in the `TailoredProfile` CRD is no longer mandatory. You can now select a list of rule objects to create a tailored profile.  Note that you must select whether your profile applies to nodes or the platform by setting the `compliance.openshift.io/product-type:` annotation or by setting the `-node` suffix for the `TailoredProfile` CR.
```

+

> * In this release, the Compliance Operator is now able to schedule scans on all nodes irrespective of the taints they have. Previously, the scan pods would only be able to tolerate the `node-role.kubernetes.io/master taint`, meaning that they would either run on nodes with no taints or only on nodes with the `node-role.kubernetes.io/master` taint. In deployments using custom taints for their nodes, this would result in the scans not being scheduled on those nodes. Now, the scan pods tolerate all node taints.

Reviewer 2: ```suggestion
* In this release, the Compliance Operator is now able to schedule scans on all nodes irrespective of their taints. Previously, the scan pods only tolerated the `node-role.kubernetes.io/master taint`, meaning that they either ran on nodes with no taints or only on nodes with the `node-role.kubernetes.io/master` taint. In deployments that use custom taints for their nodes, this resulted in the scans not being scheduled on those nodes. Now, the scan pods tolerate all node taints.
```

=== Templating and variable use

* In this release, the remediation template now allows multi-value variables. 
+
* With this update, the Compliance Operator can change remediations based on variables that are set in the compliance profile. This is useful for remediations that include deployment-specific values such as time outs, NTP server host names, or similar. Additionally, the `ComplianceCheckResult` objects now use the label `compliance.openshift.io/check-has-value` that lists the variables a check can use.

[id="openshift-compliance-operator-0-1-44-bug-fixes"]
=== Bug fixes

> * Previously, while performing a scan, an unexpected termination was occurring in one of the scanner containers of the pods. In this release, the Compliance Operator uses the latest OpenSCAP version 1.3.5 to avoid a crash.

Reviewer 2: ```suggestion
* Previously, while performing a scan, an unexpected termination occurred in one of the scanner containers of the pods. In this release, the Compliance Operator uses the latest OpenSCAP version 1.3.5 to avoid a crash.
```

+

> * Previously when you used `autoReplyRemediations` to apply remediations, this would trigger an update of the cluster nodes. This could be disruptive if some of the remediations did not include all of the required input variables. Now, if a remediation  is missing one or more required input variables, it is  assigned a state of `NeedsReview`.  If one or more remediations are in a `NeedsReview` state, the machine config pool will remain paused and the remediations will not be applied until all of the required variables are set. This helps minimize disruption to the nodes.

Reviewer 2: ```suggestion
* Previously, using `autoReplyRemediations` to apply remediations triggered an update of the cluster nodes. This was disruptive if some of the remediations did not include all of the required input variables. Now, if a remediation is missing one or more required input variables, it is assigned a state of `NeedsReview`.  If one or more remediations are in a `NeedsReview` state, the machine config pool remains paused, and the remediations will not be applied until all of the required variables are set. This helps minimize disruption to the nodes.
```

+

> * The RBAC Role and RoleBinding used for Prometheus metrics are changed to 'ClusterRole' and 'ClusterRoleBinding' to ensure that monitoring works out-of-the-box.

Reviewer 2: ```suggestion
* The RBAC Role and Role Binding used for Prometheus metrics are changed to 'ClusterRole' and 'ClusterRoleBinding' to ensure that monitoring works without customization.
```

+

> * Previously, if an error occurred while parsing a profile, rules or variables objects would be removed and deleted from the profile. Now, if an error occurs during parsing, the `profileparser` annotates the object with a temporary annotation that prevents the object from being deleted until after parsing completes. link:https://bugzilla.redhat.com/show_bug.cgi?id=1988259[(BZ#1988259)].

Reviewer 2: ```suggestion
* Previously, if an error occurred while parsing a profile, rules or variables objects were removed and deleted from the profile. Now, if an error occurs during parsing, the `profileparser` annotates the object with a temporary annotation that prevents the object from being deleted until after parsing completes. link:https://bugzilla.redhat.com/show_bug.cgi?id=1988259[(BZ#1988259)].
```

+
* Previously, an error occurred if titles or descriptions were missing from a tailored profile. Because the XCCDF standard requires titles and descriptions for tailored profiles, titles and descriptions are now required to be set in `TailoredProfile` CRs.
+ 

> * Previously, when using tailored profiles, `TailoredProfile` variable values were only allowed to be set using a specific selection set. This restriction is now removed, and `TailoredProfile` variables can be set to any value.

Reviewer 2: ```suggestion
* Previously, when using tailored profiles, `TailoredProfile` variable values were allowed to be set using only a specific selection set. This restriction is now removed, and `TailoredProfile` variables can be set to any value.
```

[id="compliance-operator-release-notes-0-1-39"]
== Release Notes for Compliance Operator 0.1.39
The following advisory is available for the OpenShift Compliance Operator 0.1.39:

* link:https://access.redhat.com/errata/RHBA-2021:3214[RHBA-2021:3214 OpenShift Compliance Operator Bug Fix and Enhancement Update]

[id="compliance-operator-0-1-39-new-features-and-enhancements"]
=== New features and enhancements

* Previously, the Compliance Operator was unable to parse Payment Card Industry Data Security Standard (PCI DSS) references. Now, the Operator can parse compliance content that ships with PCI DSS profiles.
+

> * Previously, the Compliance Operator was unable to execute rules for AU-5 control in the moderate profile. Now, permission is added to the Operator so that it can read *Prometheusrules.monitoring.coreos.com* objects and execute the rules that cover AU-5 control in the moderate profile.

Reviewer 2: ```suggestion
* Previously, the Compliance Operator was unable to execute rules for AU-5 control in the moderate profile. Now, permission is added to the Operator so that it can read *Prometheusrules.monitoring.coreos.com* objects and run the rules that cover AU-5 control in the moderate profile.
```

+ 
* You can now customize the nodes that are used to schedule the `ResultServer` workload by configuring the `nodeSelector` and `tolerations` attributes of the `ScanSetting` object. These attributes are used to place the `ResultServer` pod (a pod where the raw ARF results are uploaded to, the pod mounts a PV storage volume). Previously, the `nodeSelector` and the `tolerations` defaulted to selecting master nodes and tolerating the node-role.kubernetes.io/master taint. This did not work in environments where master nodes are not permitted to mount PVs. This feature provides another way for you to select nodes and tolerate different taints in those environments.  
+
* The Compliance Operator can now remediate `KubeletConfig` objects.
+
* A comment containing an error message is now added to help content developers differentiate between objects that do not exist in the cluster versus objects that cannot be fetched.
+
* Rule objects now contain two new attributes, `checkType` and `description`. These attributes allow you to determine if the rule pertains to a node check or platform check, and also allows you to review what the rule does.
+

> * This enhancement removes the requirement that you had to extend an existing profile in order to create a tailored profile. This means they `extends` field in the `TailoredProfile` CRD is no longer mandatory. You can now select a list of rule objects to create a tailored profile from scratch.  Note: You will need to select whether your profile applies to nodes or the platform by setting the `compliance.openshift.io/product-type:` annotation or by setting the `-node` suffix for the `TailoredProfile` CR.

Reviewer 3: s/This means they/This means the

Author: Thanks! Also added/updated the advisories.

//OpenShift Compliance Operator Release Notes
[id="compliance-operator-release-notes"]
= Compliance Operator release notes
:context: compliance-operator-release-notes-v0
include::modules/common-attributes.adoc[]

toc::[]

[id="compliance-operator-inclusive-language"]
== Making open source more inclusive

Red Hat is committed to replacing problematic language in our code, documentation, and web properties. We are beginning with these four terms: master, slave, blacklist, and whitelist. Because of the enormity of this endeavor, these changes will be implemented gradually over several upcoming releases. For more details, see link:https://www.redhat.com/en/blog/making-open-source-more-inclusive-eradicating-problematic-language[Red Hat CTO Chris Wright's message].

[id="compliance-operator-release-notes-0-1-44"]
== Release Notes for Compliance Operator 0.1.44

> Issued: 2021-

Reviewer: By the way, we hit an issue with our release pipeline which means the advisory will probably be a bit delayed. I still don't know the exact date (some of our QE colleagues are on vacation), but at least we have a bit more time to polish the relnotes

The Compliance Operator 0.1.44 is now available with enhancements and bug fixes.

[id="compliance-operator-0-1-44-new-features-and-enhancements"]
=== New features and enhancements

* To control whether scan scheduling is strict or not strict - a new `strictNode` option is available for `ComplianceScan`, `ComplianceSuite` and `ScanSetting` scans. This option defaults to true (strict), meaning that the operator will error out if a scan pod can't be scheduled. Switching the option to true makes the scan more permissive an go forward. Useful for clouds with ephemeral nodes.
+

> * In this release, the `nodeSelector` and `tolerations` in the `ScanSettings` object are now configurable  This enables deployers to configure where the Result Server will run, and thus what node will host the persistent volume that will contain the raw results. This allows us to schedule a pod that makes use of a persistent volume on the master nodes.

Reviewer: Feature: The ScanSetting object now allows the user to configure the nodeSelector and tolerations used to place the resultserver pod (a pod where the raw ARF results are uploaded to, the pod mounts a PV storage volume).

Reason: Previously, the nodeSelector and the tolerations defaulted to selecting master nodes and tolerating the `node-role.kubernetes.io/master` taint. This doesn't work in all environments, in some, master nodes are not permitted to mount any PVs. Those environments must have a way to select other nodes and tolerate different taints.

Result: Users are able to customize which nodes are used to schedule the ResultServer workload using the nodeSelector and tolerations attributes of the ScanSetting object

Author: In this case, can we substitute control plane host for master node? e.g. Previously, the nodeSelector and the tolerations defaulted to selecting control plane host and tolerating the node-role.kubernetes.io/master taint

Reviewer: I see the reasoning, but the fact is that the name of the role as well as the labels are called what they are now.
In the example you wrote, the second sentence:

This also allows you to specify the resultserver pod that mounts a PV storage volume used to store the raw Asset Reporting Format (ARF) results.

should say something like:
This also allows you to specify the nodes where the resultserver pod that mounts a PV storage volume used to store the raw Asset Reporting Format (ARF) results would be scheduled.

> * In this release, the `nodeSelector` and `tolerations` in the `ScanSettings` object are now configurable  This enables deployers to configure where the Result Server will run, and thus what node will host the persistent volume that will contain the raw results. This allows us to schedule a pod that makes use of a persistent volume on the master nodes.

Author: Resolved through rewrite above.

+

> * In this release, the Compliance Operator can now remediate `KubeletConfig` objects.

Reviewer: Do you think we should expand on this feature? Honestly, I think this feature and the variables would benefit from a subsection in the CO docs. We're still working on the upstream documentation as well, here's a draft: https://github.com/Vincent056/compliance-operator/blob/master/doc/kubeletConfig-remediations.md

+
* In this release, an enhancement had been added for content developers. An error is added to a comment for result objects that differentiates between objects that don't exist in the cluster versus objects that can't be fetched.
+

> * In this release, rule objects now contain two new attributes, `checkType` and `description`. These attributes allow the user to see if the rule pertains to a node check or to a platform check and also allows the user to audit what the rule check does.

Reviewer: Do you want me to add the feature/reason/result here as well?

Author: sure!

Reviewer: No, they are parsed from the content that comes from the CaC repository. In general the rule objects are read-only and just provide a kube-native way of representing the compliance content.

+

> * In this release, the remediation template now allows multi-value variables. 

Reviewer: I know this note was separate in the upstream relnotes as well, but since we bundled several upstream releases together, I would suggest to also bundle all notes about the templating and variable use into a single paragraph.

Author: Sure - we could add a heading or create a single note with bullets under it wherever it makes sense.

Reviewer: Thanks; I'll leave the format to you, whatever is more consistent with other projects.

Author: Thanks for your patience Jakub.
I creating a section, might be temporary,  to collect the notes about templating and variable use. 
I only have the 2 in there now. Are these right? Are there others that should be in there?

Please let me know if you see anything else that doesn't make sense or could use better wording.

+
* With this update, the Compliance Operator can change remediations based on variables that are set in the compliance profile. This is useful for remediations that include deployment-specific values such as time outs, NTP server host names, or similar. Additionally, the `ComplianceCheckResult` objects now use the label `compliance.openshift.io/check-has-value` that lists the variables a check can use.
+

> * This enhancement removes the requirement to extend an existing Profile in order to create a Tailored Profile. The `extends` field in the `TailoredProfile` CRD is no longer mandatory. You can now select a list of rule objects to create a tailor profile from scratch. Note: You will need to select whether your profile applies to nodes or the platform by setting the `compliance.openshift.io/product-type:` annotation or by setting the `-node` suffix for the `TailoredProfile` CR.

Reviewer: Good point. I think the note is still valid, even creating a tailored profile from scratch is limited by the existing selection of rules. The old tailoring functionality allowed to take a profile and enable rules that were previously disabled or disable rules that were enabled. The new functionality allows to pick and choose totally custom rules to create a new profiles.

But we should extend the chapter to include an example of creating a tailored profile from scratch. I took a note.

+

> * Previously, only master node taints were tolerated. Now, the Compliance Operator tolerates all taints, making it possible to schedule scans on all nodes.

Reviewer: Feature: The scan pods now tolerate all node taints

Reason: Previously, the scan pods would only be able to tolerate the `node-role.kubernetes.io/master` taint, meaning they would either run on nodes with no taints or on nodes with the `node-role.kubernetes.io/master` taint. However, some deployments use custom taints for their nodes, which would result in the scans not being scheduled on those nodes with custom taints.

Result: The detault tolerations were changed to "tolerate everything". Now, the CO is able to schedule scans on all nodes irrespective of the taints they have.

[id="openshift-compliance-operator-0-1-44-bug-fixes"]
=== Bug fixes

* Previously, while performing a scan, a crash was occurring in one of the scanner containers of the pods performing the scan. In this release, the Compliance Operator uses the latest OpenSCAP version 1.3.5 to avoid this issue.
+

> * Previously, only one global `KubeletConfig` object was created per pool. In this release, a `KubeletConfig` object is created for each pool. 

Author: Bug or enhancement? If bug, can you describe the issue the user would have encountered?

Author: Removed per note above.

> * Previously, only one global `KubeletConfig` object was created per pool. In this release, a `KubeletConfig` object is created for each pool. 

Reviewer: Since this is a bugfix for the kubeletConfig remediation feature that is being introduced in this update, I think we can drop this line. A customer going from 39 to 44 wouldn't ever hit the issue (the issue was only present in .43)

Author: Removed.

+

> * The rules for tailored profiles are now validated so that only rules of the same type (Platform or Node) are included in the validation.

Author: What issue was encountered?

Reviewer: You know, on second though, this was a bug in a new functionality (creating the tailored profiles from scratch). So I don't think we need to remove it separately.
The issue was that while creating a new tailored profile from scratch, at one point CO wouldn't ensure that all the rules are of same type (all node rules or all platform rules). The issue was that the rules that were of different type than the profile itself would not be applicable and it was not apparent why. So to solve that, the operator now fails with a clear error message if the user tries to create a tailored profile from scratch and some of the rules don't match the rest.

+

> * Remediations now have a `NeedsReview` state if they require a variable to be set. In order to automatically apply remediations, you must set all required variables before the MachineConfig pool is unpaused and the remediations applied.

Reviewer: Ah, sorry, you're right, this was too low level.

So CO has this option called `autoApplyRemediations`, if it's set to true (default is false), then CO will apply all the fixes for all the compliance findings, providing a sort of a one-click compliance. But because of the way the MachineConfigOperator which consumes and actually applies most of the remediations works, applying a remediation triggers an update of the cluster nodes. This is disruptive. To work around that, a MachineConfigPool (a set of machines to apply the remediations to) can be paused, all the needed remediations are marked for application, but really only applied when the pool is unpaused.

Now finally to the point of the note: In case a remediation can't be applied because some required input variable is missing (e.g. the remediation tries to set the NTP servers to be used, which can only be provided by the user), the remediation is assigned a "NeedsReview" state. As long as there are some "NeedsReview" remediations, a pool can't be unpaused.

HTH

+

> * Previously, the variable that was set for value-required handling was read from the datastream. Now, the variable is read directly from the tailoring to ensure that the appropriate remediation is created. 

Reviewer: Since this is a fix for using custom variable values which was not possible before (see the last item in this list), I think we can drop this line

Author: Removed this bullet point.

+

> * The RBAC Role and RoleBinding used for Prometheus metrics is changed to `cluster-wide` to ensure that monitoring works out-of-the-box when using ClusterRole or ClusterRoleBinding for monitoring permissions.

Author: Please double check this rewrite.

Reviewer: I would reword to:
The RBAC Role and RoleBinding used for Prometheus metrics is changed to ClusterRole and ClusterRoleBinding to ensure that monitoring works out-of-the-box.

+
* Previously, if an error occurred while parsing a profile, rules or variables objects would be removed and deleted from the profile. Now, if an error occurs during parsing, the `profileparser` annotates the object with a temporary annotation that prevents the object from being deleted until after parsing completes. link:https://bugzilla.redhat.com/show_bug.cgi?id=1988259[(BZ#1988259)].
+

> * Because the XCCDF standard discourages empty titles and descriptions for tailored profiles, titles and descriptions are now required to be set in `TailoredProfile` CRs.

Author: Did the user see an error - is that what is meant by discourages?

Reviewer: I don't remember to be honest. @JAORMX do you?

Reviewer 4: @kmccarron-rh Yes, User see an error when they create tailored profiles with titles and descriptions.

Author: Thank you @pdhamdhe - I made a slight adjustment to the language.

+ 

> * Previously, when using tailored profiles, `TailoredProfile` variable values were only allowed to be set using a specific selection set. This restriction is now removed, and `TailorProfile` variables can be set to any value.

Reviewer: thanks!

[id="compliance-operator-release-notes-0-1-39"]
== Release Notes for Compliance Operator 0.1.39
Issued: 2021-08-23

The Compliance Operator 0.1.39 is now available with enhancements.

[id="compliance-operator-0-1-39-new-features-and-enhancements"]
=== New features and enhancements

> * Previously, the Compliance Operator was unable to parse Payment Card Industry Data Security Standard (PCI DSS) references. Now, the `profileparser` can parse compliance content that ships with PCI DSS profiles.

Author: Is it important to point out `profileparser`? Or, can we just say Operator?

Reviewer: You're right, profileparser is just an implementation detail.

//OpenShift Compliance Operator Release Notes
[id="compliance-operator-release-notes"]
= Compliance Operator release notes
:context: compliance-operator-release-notes-v0
include::modules/common-attributes.adoc[]

toc::[]

> The Compliance Operator lets {product-title} administrators describe the desired compliance state of a cluster and provides them with an overview of gaps and ways to remediate them.

Reviewer 2: ```suggestion
The Compliance Operator lets {product-title} administrators describe the required compliance state of a cluster and provides them with an overview of gaps and ways to remediate them.
```

"Desire" is one of the words that we can't use because it doesn't translate well.

Author: Done
